packages/kit/src/runtime/client/renderer.js

```diff
@@ -275,14 +275,21 @@ export class Renderer {
 			this._init(navigation_result);
 		}
 
-		if (!opts?.keepfocus) {
-			document.body.focus();
+		if (opts) {
+			if (!opts.keepfocus) {
+				document.body.focus();
+			}
+
+			// Scroll to top so we can compare the `pageYOffset` below. We cannot
```

Comment:
> We cannot compare by recording the `pageYOffset` here as there is a possiblity it will change, e.g. different page heights

I don't know if this is quite true? What I meant in my last comment about using the min of `pageYOffset` and the page height is that I wonder if you could check something like `if (pageYOffset === Math.min(oldPageYOffset, newPageHeight))` to see if the user has not scrolled

Reply:
Ah great idea. I think that might work too, I'll try that when I get the time.

Reply:
That seems to work! The code looks a lot cleaner now. Pushed the changes.